### ISSUE TITLE
Standardizing GitHub Org Labels 💅

### ISSUE THREAD

**Issue by josephfusco** [wp-graphql/wp-graphql#3195]
### What problem does this address?

As a WPGraphQL maintainer, I'd like to have consistent labels across repos in order for project management to be more streamlined.

### What is your proposed solution?

I'm proposing the following standards for the WPGraphQL org:

- all org labels are lowercase
- emojis in labels should always be after the text
- utilize the "taxonomy: term" pattern for improved categorization
- generally consistent labels across repos

## Global Labels

- `#ffffff` wontfix - *This will not be worked on*
- `#fcbfe3` good first issue - *Issue that doesn't require previous experience with codebase*
- `#e03590` close candidate - *Needs confirmation before closing*
- `#adb5bd` duplicate - *This issue or pull request already exists* (Changed from gray to a lighter gray for better visibility)
- `#6c757d` invalid? - *Something is not right here* (Changed to a darker gray to distinguish from other grays)
- `#ced4da` stale? - *May need to be revalidated due to prolonged inactivity* (Changed to an even lighter gray to differentiate within grays)
- `#d93f0b` regression - *Bug that causes a regression to a previously working feature*
- `#0e8a16` help wanted - *Issue with a clear description that the community can help with*

## Has

- `#adb5bd` has: workaround - *A temporary workaround has been provided* (Changed to a uniform gray to match other informational labels)

## Compatibility

- `#c82333` compat: breaking change - *This is a breaking change to existing functionality* (Brightened red for urgency)
- `#e0a800` compat: possible break - *There is a possibility that this might lead to breaking changes, but not confirmed yet* (Changed from a muted red to a clear yellow for caution)

## Level of Effort

- `#dc3545` effort: high - *More than a week* (Brightened to a more urgent red)
- `#ffc107` effort: med - *Less than a week* (Changed to a standard yellow for moderate effort)
- `#28a745` effort: low - *Around a day or less* (Standard green for low effort)

## Level of Impact

- `#28a745` impact: high - *Unblocks new use cases, substantial improvement to existing feature, fixes a major bug* (Uniform green for positive impact)
- `#ffc107` impact: med - *Minor performance improvements, fix broad user base issues* (Yellow for moderate impact)
- `#f8d7da` impact: low - *Fixes a minor issue for some people, slight DX improvement* (Soft pink for less critical improvements)

# Language

- `#45229e` lang: php - *Pull requests that update PHP code*
- `#168799` lang: javascript - *Pull requests that update JavaScript code*

## Needs Something

- `#6c757d` needs: discussion - *Requires a discussion to proceed* (Changed to a darker gray for visibility and importance)
- `#adb5bd` needs: info - *More information is needed to resolve this issue* (Uniform gray for informational needs)
- `#ced4da` needs: reproduction - *This issue needs to be reproduced independently* (Lighter gray to distinguish from more urgent needs)
- `#f2994a` needs: reviewer response - *This needs the attention of a codeowner or maintainer*
- `#56ccf2` needs: author response - *Pending information from the author*
- `#adb5bd` needs: tests - *Tests should be added to be sure this works as expected* (Uniform gray for consistency)

## Issue Scope

- `#b0c8d9` scope: build scripts - *Automating task runners and compilation processes*
- `#b8a8d6` scope: code quality - *Refactoring, linting, and enforcing coding standards*
- `#a8d2a0` scope: dependencies - *Managing, updating, or removing dependencies*
- `#ded6b0` scope: docs - *Updating, correcting, and improving documentation*
- `#a8d6d4` scope: extensions - *Integrating plugins, add-ons, or other extensions*
- `#ded3a0` scope: i18n - *Internationalizing, translating, and localizing*
- `#a8dbd4` scope: performance - *Enhancing speed and efficiency*
- `#deb5a0` scope: security - *Securing against vulnerabilities and threats*
- `#d3b8d6` scope: tests - *Developing unit tests, integration tests, and ensuring coverage*
- `#5ac5c5` scope: accessibility - *Enhancing accessibility and ensuring compliance with WCAG/ADA standards*
- `#b8a6d9` scope: graphiql - *Issues related to the GraphiQL interface enhancements or issues.*

## Statuses

- `#28a745` status: actionable - *Ready for work to begin* (Reverted to green for intuitive 'go' signaling)
- `#dc3545` status: blocked - *Progress halted due to dependencies or issues* (Bright red for critical blockage)
- `#fd7e14` status: in progress - *Currently being worked on* (Changed to a vibrant orange for visibility)
- `#17a2b8` status: in review - *Awaiting review before merging or closing* (Changed to a soothing blue to indicate ongoing review)

## Issue Type

- `#d73a4a` type: bug - *Issue that causes incorrect or unexpected behavior*
- `#fff1bc` type: chore - *Maintenance tasks, refactoring, and other non-functional changes*
- `#84b6eb` type: enhancement - *Improvements to existing functionality*
- `#f9ab45` type: feature - *New functionality being added*
- `#cc317c` type: question - *An issue that involve inquiries, clarifications, or requests for guidance*
- `#bfdadc` type: spike - *An issue that needs more research before becoming actionable*
- `#e88bdd` type: release - *Pull request intended for a release*
- `#005d5d` type: epic - *Large-scale feature or initiative that spans multiple issues and pull requests.*

## Dependencies

These should use the associated branding color

- `#7f54b3` dep: woocommerce - *Integration or compatibility with WooCommerce*
- `#c9471f` dep: wpml - *Related to WPML for multilingual support*
- `#b2fe80` dep: acf - *Involving Advanced Custom Fields functionality*
- `#a8c0d6` dep: polylang - *Integrating Polylang for multilingual support*

## Components

Specific components of the application

- `#fce6c4` component: ________ - *Relating to ________*

### wp-graphql

- `#59f7ad` not stale - *Short circuit stalebot. USE SPARINGLY.*

### What alternatives have you considered?

Crying from cognitive overload

<img src="https://media0.giphy.com/media/IplvQCq2ZdwPUGcaVN/giphy.gif"/>

**Comment by jasonbahl** [comment]
@justlevine @kidunot89 - @josephfusco and I went through the labels on the following repos to bring some standardization, following the above label definitions

- wp-graphql
- wp-graphql-smart-cache
- wpgraphql-acf
- wpgraphql.com
- acf.wpgraphql.com

Wanted to capture this here so ya'll know why some labels might have changed on you. 

Feel free to share feedback if you feel strongly about a label name, color, description, etc.

**Comment by jasonbahl** [comment]
**Note**: we do not have any automation around this and no immediate plans to automate it. . .so for the moment keeping labels in sync across repos will be a manual process.   Some repos have labels specific to the project, but for common labels like those listed above we wanted to bring some consistency.

**Comment by justlevine** [comment]
@jasonbahl please tell me you're using a script for this and not manually updating the tags and colors 😅

Few thoughts (because of _course_ I do):
- `wontfix`: seems better a `status:` taxonomy (all the other 'Globals' add context to other labels, but `wontfix` describes the current status of work related to the ticket as _Not being worked on because it's out of scope_ (for whatever reasons).
- `type: spike`: also seems better as a `status:`. bugs, enhancements, etc can all be "spiked" but it doesn't change the type of issues being discussed.
- `type: epic` personally I'd avoid PM jargon and call this `type:tracking`, but 🤷
- `deps: *` : What are the purposes/uses for these?
   - issues related to Org extensions should be transferred to those repos.
   - issues related to community extensions should be closed since they're beyond the scope of this repo.
   - Any "repo-valid" issues either affect more than one plugin, or are likely 'one-off', making this taxonomy either unhelpful or unscalable.
   
   Instead I'd recommend either reusing `scope: extensions` or adding a `scope: compatibility`.

- Effort / Impact labels: I wonder if these are actually helpful or primarily visual noise? Since they're subjective, we don't usually spend the time evaluating these during initial triage. If they're more for the PM/planning side, it would probably save you/joe the headache to just have them as custom fields in the project (at least until you're using label->field mapping automation).

- `not stale`: Please no 🙏🙏🙏😅 Short-circuiting stalebot just tries to workaround the fact that stalebot is implemented in a way that isn't a good fit for the repo, and causes more problems than both it and stalebot as a whole solve. (It's also a lie, there is _no such thing_ as a ticket that cannot go stale ;) )

   Instead I recommend we fully embrace the use of a semantic `stale?`:
   1. Stalebot _never_ autocloses a ticket. It's too stupid to know if something isn't getting worked on because it's a trash ticket or because we rely on open source contribs. Instead it only ever marks things as `stale?`
   2. `stale?` is semantic indicator that significant time has passed since there was last activity on the ticket, and it _might_ be out of date. There is no deadline imposed by a robot dictator to artificially inflate activity or risk closure. 
   3. Instead, you remediate a `stale?` ticket by... doing nothing until you're ready to work on the ticket. Then you just double check your assumptions and proceed as usual.
   4. `stale?` tickets can still be gardened as desired, either by closing the ticket or revalidating the issue.

**Comment by jasonbahl** [comment]
@justlevine thanks for the thorough feedback!  🙏🏻 

Some thoughts: 

> wontfix: seems better a status

I think I'm fine with this change. 

> type: spike: also seems better as a status

🤔 I guess in my mind a Spike is a "type" of issue that isn't actually intended to lead directly to code. It's intended to initiate some further research and investigation which should then lead to other types of issues (i.e. bugs, enhancements, features, etc)

I think I _probably_ still consider a Spike a type of issue. . . then the Spike can have a status like: `actionable`, `blocked` etc. 


> type: epic personally I'd avoid PM jargon and call this type:tracking, but 🤷

I think we'll probably be able to remove this with some of the work we've been exploring with Github projects. 

My thought process was that we sometimes like to create "Epic" issues with bullet lists to help us brainstorm, but don't have the time to immediately convert al the bullet lists into detailed issues, so labeling the issue as "type: epic" can be helpful until we're able to take the time to break the brainstorm list into individual issues which would then ideally be tagged with the appropriate Epic (or equivelent) field in the project board. 

> deps: * : What are the purposes/uses for these?

partly for use cases like maintainers or users finding related issues, i.e. (https://github.com/wp-graphql/wp-graphql/issues?q=is%3Aissue+label%3A%22dep%3A+wpml%22+is%3Aclosed)

> Effort / Impact labels: I wonder if these are actually helpful or primarily visual noise? 

We can explore leaving these on the project board exclusively. 

> not stale: Please no

🤣 

You got it, dude! We can remove this. I think this slipped through because the description made it seem like it was still tied to an action, but I recall us fixing the stalebot a while ago.

**Comment by justlevine** [comment]
What's the difference between `Spike` and `Epic` (or a theoretical `Idea`)?

In my mind these can all fall neatly into a `type: tracking` that makes sense semantically for all those cases: tracking a decision, tracing a check list, tracking other issues, tracking prior art for a potential idea, tracking cross-repo issue cleanup etc).

If you see the benefit for differentiation, go ahead.  But maybe consider a different name for spike or a clearer description to differentiate it as a 'type' during triage, because usually it means 'this issue (whatever the type is) will be dealt with in a future release).

> but I recall us fixing the stalebot a while ago.

We did, but like the issues is closes and other horror stories, these things have a tendency of coming back to life if you're not vigilant 🧟‍♂️🧟‍♂️🧟‍♂️
